File: ToolKit/src/main/java/com/uday/android/toolkit/ui/CustomToast.java
package com.uday.android.toolkit.ui;
import android.widget.*;
import android.content.*;
import android.app.*;
import com.uday.android.toolkit.R;
import android.graphics.*;

public class CustomToast extends Toast
{
	private Context context;
	
	public CustomToast(Context context){
		super(context);
		this.context=context;
	}
	
	public static void showSuccessToast(final Context context,final String message,final int duration){
		((Activity)context).runOnUiThread(new Runnable(){
			@Override
			public void run(){
				LinearLayout toastLayout=(LinearLayout)((Activity)context).getLayoutInflater().inflate(R.layout.toast_custom,null);
				TextView text=(TextView)toastLayout.findViewById(R.id.toast_text);
				text.setText(message);
				toastLayout.setBackground(context.getResources().getDrawable(R.drawable.toast_success));
				Toast toast=new Toast(context);
				toast.setDuration(duration);
				toast.setView(toastLayout);
				toast.show();
			}
		});
	}
	
	public static void showFailureToast(final Context context,final String message,final int duration){
		((Activity)context).runOnUiThread(new Runnable(){
				@Override
				public void run(){
					LinearLayout toastLayout=(LinearLayout)((Activity)context).getLayoutInflater().inflate(R.layout.toast_custom,null);
					TextView text=(TextView)toastLayout.findViewById(R.id.toast_text);
					text.setText(message);
					toastLayout.setBackground(context.getResources().getDrawable(R.drawable.toast_failure));
					Toast toast=new Toast(context);
					toast.setDuration(duration);
					toast.setView(toastLayout);
					toast.show();
				}
			});
	}
	
	public static void showNotifyToast(final Context context,final String message,final int duration){
		((Activity)context).runOnUiThread(new Runnable(){
				@Override
				public void run(){
					LinearLayout toastLayout=(LinearLayout)((Activity)context).getLayoutInflater().inflate(R.layout.toast_custom,null);
					TextView text=(TextView)toastLayout.findViewById(R.id.toast_text);
					text.setText(message);
					toastLayout.setBackground(context.getResources().getDrawable(R.drawable.toast_notify));
					Toast toast=new Toast(context);
					toast.setDuration(duration);
					toast.setView(toastLayout);
					toast.show();
				
				}
			});
		}
}

File: app/src/main/java/com/uday/android/toolkit/ui/CustomToast.kt
package com.uday.android.toolkit.ui

import android.annotation.SuppressLint
import android.content.Context
import android.widget.LinearLayout
import android.widget.TextView
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import com.uday.android.toolkit.R

class CustomToast(private val context: Context) : Toast(context) {
    companion object {

        @SuppressLint("InflateParams")
        fun showSuccessToast(context: Context, message: String, duration: Int) {
            (context as AppCompatActivity).runOnUiThread {
                val toastLayout =
                    context.layoutInflater.inflate(R.layout.toast_custom, null) as LinearLayout
                val text = toastLayout.findViewById(R.id.toast_text) as TextView
                text.text = message
                toastLayout.background =
                    context.getResources().getDrawable(R.drawable.toast_success)
                val toast = Toast(context)
                toast.duration = duration
                toast.view = toastLayout
                toast.show()
            }
        }

        @SuppressLint("InflateParams")
        fun showFailureToast(context: Context, message: String, duration: Int) {
            (context as AppCompatActivity).runOnUiThread {
                val toastLayout =
                    context.layoutInflater.inflate(R.layout.toast_custom, null) as LinearLayout
                val text = toastLayout.findViewById(R.id.toast_text) as TextView
                text.text = message
                toastLayout.background =
                    context.getResources().getDrawable(R.drawable.toast_failure)
                val toast = Toast(context)
                toast.duration = duration
                toast.view = toastLayout
                toast.show()
            }
        }

        @SuppressLint("InflateParams")
        fun showNotifyToast(context: Context, message: String, duration: Int) {
            (context as AppCompatActivity).runOnUiThread {
                val toastLayout =
                    context.layoutInflater.inflate(R.layout.toast_custom, null) as LinearLayout
                val text = toastLayout.findViewById(R.id.toast_text) as TextView
                text.text = message
                toastLayout.background = context.getResources().getDrawable(R.drawable.toast_notify)
                val toast = Toast(context)
                toast.duration = duration
                toast.view = toastLayout
                toast.show()
            }
        }
    }
}
